traits/tests/test_ctraits.py

```diff
@@ -135,25 +123,63 @@ def test_is_mapped(self):
 
         self.assertTrue(trait.is_mapped)
 
-    def test_no_value_test(self):
+    def test_default_comparison_mode(self):
         trait = CTrait(TraitKind.trait)
 
-        self.assertFalse(trait.no_value_test)
-
-        trait.comparison_mode(ComparisonMode.no_compare)
-
-        self.assertTrue(trait.no_value_test)
-
-        with self.assertRaises(AttributeError):
-            trait.no_value_test = False
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.equality_compare
+        )
 
     def test_invalid_comparison_mode(self):
         trait = CTrait(TraitKind.trait)
 
         # comparison modes other than {0,1,2}
         # are invalid
         with self.assertRaises(ValueError):
-            trait.comparison_mode(-1)
+            trait.comparison_mode = -1
 
         with self.assertRaises(ValueError):
-            trait.comparison_mode(3)
+            trait.comparison_mode = 3
+
+    def test_comparison_mode_int(self):
+        trait = CTrait(TraitKind.trait)
+
+        trait.comparison_mode = 0
+
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.no_compare
+        )
+
+        trait.comparison_mode = 1
+
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.object_id_compare
+        )
+
+        trait.comparison_mode = 2
+
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.equality_compare
+        )
+
+    def test_comparison_mode_enum(self):
+        trait = CTrait(TraitKind.trait)
+
+        trait.comparison_mode = ComparisonMode.no_compare
+
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.no_compare
+        )
+
+        trait.comparison_mode = ComparisonMode.object_id_compare
+
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.object_id_compare
+        )
+
+        trait.comparison_mode = ComparisonMode.equality_compare
+
+        self.assertEqual(
+            trait.comparison_mode, ComparisonMode.equality_compare
+        )
+
```

Comment:
Nitpick: blank line at the end of the file. `flake8` complains about this, or at least it will, once master is merged into this branch :-). (Tip: you may be able to configure your editor/IDE to automatically remove such lines. E.g., VS Code has a `files.trimFinalNewlines` setting.)